compiler/src/dotty/tools/dotc/config/CliCommand.scala

```diff
@@ -130,7 +132,8 @@ trait CliCommand:
       report.echo(ifErrorsMsg)
       None
     else if settings.version.value then
-      report.echo(versionMsg)
+      // ctx defaults to Console.err
+      ConsoleReporter(writer = new PrintWriter(Console.out, true)).printMessage(versionMsg)
```

Comment:
I think this should be handled by the reporter instead, then in the ConsoleReporter you can add an extra writer for where echo messages go (set to `Console.out` by default), and intercept `Info` diagnostics in `doReport`. This is what Scala 2 does.

Typically we do want `echo` messages to go be visible in the same UI (e.g. in an IDE so you get the nice blue info messages).